rct229/rulesets/ashrae9012019/section19/section19rule13.py

```diff
@@ -0,0 +1,255 @@
+            each_rule=Section19Rule13.HVACRule(),
+            index_rmr="baseline",
+            id="19-13",
+            description="For baseline system types 1-8 and 11-13, system design supply airflow rates shall be based on a supply-air-to-room temperature set-point difference of 20°F or the minimum outdoor airflow rate, or the airflow rate required to comply with applicable codes or accreditation standards, whichever is greater. For systems with multiple zone thermostat setpoints, use the design set point that will result in the lowest supply air cooling set point or highest supply air heating set point.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.8.1 and Exception 1",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+
+        zone_info = {}
+        for hvac_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b:
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                zone_info[zone_id_b] = {}
+                zone_info[zone_id_b][
+                    "are_all_hvac_sys_fan_objs_autosized"
+                ] = are_all_hvac_sys_fan_objs_autosized(rmi_b, hvac_id_b)
+                zone_info[zone_id_b]["proposed_supply_airflow"] = ZERO.FLOW
+                zone_info[zone_id_b][
+                    "all_design_setpoints_delta_Ts_are_per_reqs"
+                ] = ZERO.TEMPERATURE
+
+                # get `design_thermostat_cooling_setpoint` and `design_thermostat_heating_setpoint`
+                for zone_b in find_all(
+                    '$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")]',
+                    rmi_b,
+                ):
+                    zone_info[zone_id_b]["design_thermostat_cooling_setpoint"] = min(
+                        find_all(
+                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")].design_thermostat_cooling_setpoint',
+                            rmi_b,
+                        )
+                    )
+                    zone_info[zone_id_b]["design_thermostat_heating_setpoint"] = max(
+                        find_all(
+                            f'$.buildings[*].building_segments[*].zones[?(@.id == "{zone_id_b}")].design_thermostat_heating_setpoint',
+                            rmi_b,
+                        )
+                    )
+
+                    for space_b in getattr_(zone_b, "Zone", "spaces"):
```

Comment:
can be simplified to 
```python
zone_has_lab_space = any(space_b.get("lighting_space_type") == LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM) for space in find_all("$.spaces[*]", zone_b)]
```
Note:
1. We don't want to fail the missing `lighting_space_type` because some space may not have `lighting_space_type` - e.g., attic. In that case, we shall handle it as `False`.
2. We do not want to fail a zone with no space in here -> the any([]) returns `False`

The simplified code handles both cases with `False` outcome.

Reply:
Good point! Addressed.